### PR TITLE
Tracking slow resource / code

### PR DESCRIPTION
**Pull request by joelwurtz**
Will not say this is perfect but it's a start to know what is slow

### CODE REVIEW

File: data/shared/citizen/scripting/lua/scheduler.lua [hunk 1]
```diff
@@ -21,17 +27,32 @@ CreateThread = Citizen.CreateThread
...
 
 	-- add new thread and save old thread
 	local oldThread = curThread
 	curThread = t
 
+	local startAt = GetGameTimer()
 	local result, err = coroutine.resume(coro)
+	local time = GetGameTimer() - startAt
+
+	if time > 10 then
```
Comment (joelwurtz) [inline data/shared/citizen/scripting/lua/scheduler.lua]: Not sure about this limit

**Comment by joelwurtz** [comment]
An example of what we get after this 

slow-example

File: data/shared/citizen/scripting/lua/scheduler.lua (hunk 1 above)
Comment (Hellslicer) [reply data/shared/citizen/scripting/lua/scheduler.lua]: `>= 150` for consistency with the underneath hitch warning?

Comment (joelwurtz) [reply data/shared/citizen/scripting/lua/scheduler.lua]: Not really a hitch warning is for the whole frame where there can be many threads executed here

Comment (Hellslicer) [reply data/shared/citizen/scripting/lua/scheduler.lua]: IMO something between 50 and 100 would be fine, 10ms is way too low

**Comment by minipunch** [comment]
Why was this not merged? I am currently trying to find the source of some hitch warnings I am getting and this seems like it would be a helpful change.

**Comment by Jarrrk** [comment]
@minipunch because a better resource monitor was implemented - In the F8 console type resmon 1

**Comment by vecchiotom** [comment]
this is serverside though @Jarrrk

**Comment by PichotM** [comment]
probably not pushed since it may impacts client performance
and if you want to use it, you can just edit your sheduler ^_^

**Comment by minipunch** [comment]
@PichotM True.. I might just edit the scheduler manually to use it. What makes you think it might impact client performance?

**Comment by sirarrowhead** [comment]
@minipunch I did it a long time ago, i have not noticed any worse peformance

**Pull request closed without merging**